src/V3Const.cpp

```diff
@@ -242,6 +655,64 @@ class ConstVisitor final : public AstNVisitor {
         VL_DO_DANGLING(nodep->deleteTree(), nodep);
         return true;
     }
+    bool matchBitOpTree(AstNode* nodep) {
+        AstNode* newp = nullptr;
+        bool tried = false;
+        if (AstAnd* andp = VN_CAST(nodep, And)) {  // 1 & BitOpTree
+            if (AstConst* bitMaskp = VN_CAST(andp->lhsp(), Const)) {
+                if (bitMaskp->num().toUQuad() != 1) return false;
+                newp = ConstBitOpTreeVisitor::simplify(andp->rhsp(), 1, m_statBitOpReduction);
+                tried = true;
+            }
+        }
+        if (!tried) {
+            // (comp == BitOpTree) & BitOpTree
+            // (comp != BitOpTree) | BitOpTree
+            newp = ConstBitOpTreeVisitor::simplify(nodep, 0, m_statBitOpReduction);
+        }
+        if (newp) {
+            UINFO(4, "Transformed leaf of bit tree to " << newp << std::endl);
+            if (debug() >= 9) {
+                static int c = 0;
+                std::cout << "Call matchBitOpTree[" << c << "]\n";
+                nodep->dumpTree(std::cout);
+                std::cout << "\nResult:\n";
+                newp->dumpTree(std::cout);
+                ++c;
+            }
+            nodep->replaceWith(newp);
+            VL_DO_DANGLING(nodep->deleteTree(), nodep);
+        }
+        return newp;
+    }
+    bool matchRedundantClean(AstAnd* andp) {
```

Comment:
I think this optimization is separable from the rest, can you pull it out and on pass merge it?

If we get "customer" problems with this patch once in mainline, being able to separate out some of the possible causes will be helpful. 

Also if there's any other optimizations other than the biggest that are separable please do.

Thanks